crates/next-custom-transforms/src/transforms/server_actions.rs

```diff
@@ -112,31 +117,32 @@ struct ServerActions<C: Comments> {
 
     annotations: Vec<Stmt>,
     extra_items: Vec<ModuleItem>,
+    hoisted_extra_items: Vec<ModuleItem>,
     export_actions: Vec<String>,
 
     private_ctxt: SyntaxContext,
 }
 
 impl<C: Comments> ServerActions<C> {
     // Check if the function or arrow function is an action function
-    fn get_action_info(
-        &mut self,
-        maybe_body: Option<&mut BlockStmt>,
-        remove_directive: bool,
-    ) -> bool {
+    fn get_action_info(&mut self, maybe_body: Option<&mut BlockStmt>) -> (bool, bool) {
```

Comment:
maybe time for a rename? it no longer only checks for actions
(also i'd consider using a struct `{ is_cache_fn: bool, is_action_fn: bool }` for the result, the tuple feels a bit cryptic)